In [1]:
import pandas as pd
import numpy as np



In [2]:
data=pd.read_csv('dataset w7.csv')

In [3]:
data


Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

# 1. Carry out descriptive analytics on the dataset.


In [4]:
data.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

# 2. Check whether female and male students scored the same marks.


In [5]:
import scipy.stats as stats
stats.ttest_ind(data['Total Marks'][data['Gender'] == 'male'],
                 data['Total Marks'][data['Gender'] == 'female'])

Ttest_indResult(statistic=-0.6940885268848668, pvalue=0.4937822317960763)

solution

take Null Hypothesis Ho as 'the marks scored by male and female students are not same'
   
   df=n-1 = 28-1 = 27
   level of significance is 5% = 0.05
   
  then the table value is 2.052
  
  the calculated value is -0.694
  
  Here, |t score|=0.694 < 2.052 then accept the null hypothesis 
  
  Also, p value=0.49>0.05 implies accept Ho
  
  which shows that the marks of female and male students were not same


# 3. Whether test preparation helps the students?


In [7]:
import scipy.stats as stats
stats.ttest_ind(data['Total Marks'][data['Test preparation'] == 'none'],
                 data['Total Marks'][data['Test preparation'] == 'completed'])

Ttest_indResult(statistic=-3.6143637527769217, pvalue=0.0012671292457320106)

solution

lets take the Null Hypothesis  Ho as 'the Test preparation does not helps the students'
then the alternative hypothesis H1 is 'the test preparation helps the students'

df=n-1 = 28-1 =27
level of significance =5% = 0.05
the corresponding table value = 2.052

calculated value = -3.614
Here ,|t score|=3.614 > 2.052 , then reject Ho and accept alternative hypothesis

Also,p value = 0.00126 < 0.05 implies reject Ho and accept H1 

which shows that the test preparation helps the students
